R/target_sda.R

```diff
@@ -130,14 +130,25 @@ target_sda <- function(data,
   ald_by_sector <- ald %>%
     aggregate_excluding(c("technology", "plant_location", "country_of_domicile"))
 
-  loanbook_with_weighted_emission_factors <- calculate_weighted_emission_factor(
-    data,
-    ald = ald_by_sector,
-    use_credit_limit = use_credit_limit,
-    by_company = by_company
-  )
+  data <- inner_join(data, ald_by_sector, by = ald_columns())
+
+  if (by_company) {
+    data <- data %>%
+      summarize_weighted_emission_factor(
+        "name_ald",
+        use_credit_limit = use_credit_limit
+      )
+  } else {
+    data <- data %>%
+      summarize_weighted_emission_factor(
+        use_credit_limit = use_credit_limit
+      )
+    }
```

Comment:
I moved `summarize_weighted_emission_factor` to `./R/summarize_weighted_production.R`, and adjusted it's interface to reflect that of the other `summarize_weighted_*` functions